src/Psalm/Internal/Analyzer/Statements/Expression/CastAnalyzer.php

```diff
@@ -300,6 +289,444 @@ public static function analyze(
         return false;
     }
 
+    public static function castIntAttempt(
+        StatementsAnalyzer $statements_analyzer,
+        Context $context,
+        Union $stmt_type,
+        PhpParser\Node\Expr $stmt,
+        bool $explicit_cast = false
+    ): Union {
+        $codebase = $statements_analyzer->getCodebase();
+
+        $risky_cast = [];
+        $invalid_casts = [];
+        $valid_ints = [];
+        $castable_types = [];
```

Comment:
Not sure why there is both `$valid_ints` and `$castable_types`, is it just organizational because `$valid_ints` only holds literals? Mind adding a short comment?